packages/devtools_app_shared/lib/src/service/service_extension_manager.dart

```diff
@@ -362,41 +366,45 @@ final class ServiceExtensionManager with DisposerMixin {
 
       assert(value != null);
-      if (value is bool) {
-        Future<void> call(String? isolateId, bool value) async {
+      try {
+        if (value is bool) {
+          Future<void> call(String? isolateId, bool value) async {
+            await _service!.callServiceExtension(
+              name,
+              isolateId: isolateId,
+              args: {'enabled': value},
+            );
+          }
+
+          final description = extensions.serviceExtensionsAllowlist[name];
+          if (description?.shouldCallOnAllIsolates ?? false) {
+            // TODO(jacobr): be more robust instead of just assuming that if the
+            // service extension is available on one isolate it is available on
+            // all. For example, some isolates may still be initializing so may
+            // not expose the service extension yet.
+            await _service!.forEachIsolate((isolate) async {
+              await call(isolate.id, value);
+            });
+          } else {
+            await call(mainIsolate?.id, value);
+          }
+        } else if (value is String) {
           await _service!.callServiceExtension(
             name,
-            isolateId: isolateId,
-            args: {'enabled': value},
+            isolateId: mainIsolate?.id,
+            args: {'value': value},
+          );
+        } else if (value is double) {
+          await _service!.callServiceExtension(
+            name,
+            isolateId: mainIsolate?.id!,
+            // The param name for a numeric service extension will be the last part
+            // of the extension name (ext.flutter.extensionName => extensionName).
+            args: {name.substring(name.lastIndexOf('.') + 1): value},
           );
         }
-
-        final description = extensions.serviceExtensionsAllowlist[name];
-        if (description?.shouldCallOnAllIsolates ?? false) {
-          // TODO(jacobr): be more robust instead of just assuming that if the
-          // service extension is available on one isolate it is available on
-          // all. For example, some isolates may still be initializing so may
-          // not expose the service extension yet.
-          await _service!.forEachIsolate((isolate) async {
-            await call(isolate.id, value);
-          });
-        } else {
-          await call(mainIsolate?.id, value);
-        }
-      } else if (value is String) {
-        await _service!.callServiceExtension(
-          name,
-          isolateId: mainIsolate?.id,
-          args: {'value': value},
-        );
-      } else if (value is double) {
-        await _service!.callServiceExtension(
-          name,
-          isolateId: mainIsolate?.id!,
-          // The param name for a numeric service extension will be the last part
-          // of the extension name (ext.flutter.extensionName => extensionName).
-          args: {name.substring(name.lastIndexOf('.') + 1): value},
-        );
+      } on RPCError {
+        return;
```

Comment:
should we be logging this error anywhere instead of just swallowing it? In what case do we expect an RPCError to be thrown?

Reply:
In theory, any of the codes listed [here](https://pub.dev/documentation/vm_service/latest/vm_service/RPCErrorKind.html) could be returned, but in general, only those with the form `-32xxx` will be returned for service extensions.

We could make this only catch exceptions with code `-32000`, which will cover the cases where the connection disappears or is disposed of.